server/modules/environment-state/getServiceTargetStates.js

```diff
@@ -2,93 +2,78 @@
-   
   var checksGrouped = _.groupBy(serviceChecks, 'Name');
   return _.map(checksGrouped, function (checks, checkName) {
-     
     // If some instances checks failed for a given check, mark as failed
     // also, don't count in instance into working
-    if (_.some(checks, { Status: 'critical' })) {
-      return {
-        Name: checks[0].Name,
-        Status: Enums.HEALTH_STATUS.Error,
-      };
-    } else {
-      return {
-        Name: checks[0].Name,
-        Status: Enums.HEALTH_STATUS.Healthy,
-      };
-    }
-  });
-}
-
-function getServiceOverallHealth(healthChecks, instances) {
-  if (_.some(healthChecks, { Status: Enums.HEALTH_STATUS.Error })) {
     return {
-      Status: Enums.HEALTH_STATUS.Error,
+      Name: checks[0].Name,
+      Status: _.some(checks, { Status: 'critical' }) ? HEALTH_GOOD : HEALTH_BAD
     };
-  } else {
-    return {
-      Status: Enums.HEALTH_STATUS.Healthy
-    };
-  }
+  });
 }
 
-function* getServicesState(environmentName, runtimeServerRoleName, instances) {
-  let servicesList = yield serviceReporter.getServicesList(environmentName, runtimeServerRoleName);
+function getServiceOverallHealth(healthChecks) {
+  return {
+    Status:_.some(healthChecks, { Status: HEALTH_BAD }) ? HEALTH_BAD : HEALTH_GOOD
+  };
+}
 
+function* getServicesTargetState(environmentName, runtimeServerRoleName, instances) {
+  let targetServiceStates = yield targetStates.getAllServiceTargets(environmentName, runtimeServerRoleName);
   let allServiceObjects = _.flatMap(instances, instance => instance.Services);
 
   // Find all objects representing particular service for all nodes
   let servicesGrouped = _.groupBy(allServiceObjects, 'Name');
-  let services = _.map(servicesList, (service) => {
 
+  return _.map(targetServiceStates, (service) => {
     // serviceObjects now has all 'Name' service descriptors in instances
     let serviceObjects = servicesGrouped[service.Name];
     _.each(serviceObjects, (obj) => {
-      if (obj.Version !== service.Version) {
-        logger.warn(`${service.Name} ${service.Version} TODO service versions dont match - implement handling`);
-      }
-      if (obj.DeploymentId !== service.DeploymentId) {
-        logger.warn(`${service.Name} ${service.Version} TODO deployment ids dont match - implement handling`);
-      }
+      checkServiceProperties(obj, service, 'Version');
+      checkServiceProperties(obj, service, 'DeploymentId');
     });
 
     let serviceInstances = _.filter(instances, instance => _.some(instance.Services, { Name: service.Name }));
-
     let healthyNodes = _.filter(serviceInstances, (instance) => instance.OverallHealth.Status === Enums.HEALTH_STATUS.Healthy);
     let instancesHealthCount = healthyNodes.length + '/' + serviceInstances.length;
-
-    let serviceHealthChecks = getServiceChecksInfo(serviceObjects)
+    let serviceHealthChecks = getServiceChecksInfo(serviceObjects);
+    let Status = service.hasOwnProperty('Status') ? service.Status : 'Enabled';
```

Comment:
not sure if that's the best name for enabled / disabled service, status could also point at AWS instance status, might be confusing, especially if we add this to instance